src/nodemanagerbase.cpp

```diff
@@ -224,185 +226,195 @@ UA_LocalizedText make_localised( UaString text )
-                                                parent->nodeId().impl(),
-                                                refType.impl(),
-                                                to->browseName().impl(),
-                                                UaNodeId(UA_NS0ID_BASEDATAVARIABLETYPE ,0).impl() ,
-                                                attr,
-                                                dateDataSource,
-                                                static_cast<void*>(to),  // this is our nodeContext - we'll use it to map to the variable
-                                                nullptr
-                                               );
-        if (UA_STATUSCODE_GOOD == s)
-        {
-            m_listNodes.push_back( to );
-            parent->addReferencedTarget( to, refType );
-        }
-
-
-        return s;
+        m_listNodes.push_back( to );
+        parent->addReferencedTarget( to, refType );
     }
+    return s;
+}
 
-    case OpcUa_NodeClass_Method:
-    {
-        UA_MethodAttributes attr;
-        UA_MethodAttributes_init(&attr);
-        attr.executable = true;
-        attr.userExecutable = true;
-        attr.displayName = *displayName.impl();
-        attr.description = *dummyDescription.impl();
+UaStatus NodeManagerBase::addMethodNodeAndReference(
+    UaNode* parent,
+    UaNode* to,
+    const UaNodeId& refType)
+{
+    UaLocalizedText displayName( "en_US", to->browseName().unqualifiedName().toUtf8().c_str());
+    UA_MethodAttributes attr;
+    UA_MethodAttributes_init(&attr);
+    attr.executable = true;
+    attr.userExecutable = true;
+    attr.displayName = *displayName.impl();
+    attr.description = *emptyDescription.impl();
 
-        MethodHandleUaNode *handle = new MethodHandleUaNode;
-        handle->setUaNodes( static_cast<UaObject*>(parent), static_cast<UaMethod*>(to) );
+    MethodHandleUaNode *handle = new MethodHandleUaNode;
+    handle->setUaNodes( static_cast<UaObject*>(parent), static_cast<UaMethod*>(to) );
 
-        LOG(Log::TRC) << "parent node: " << parent->nodeId().toFullString().toUtf8();
+    LOG(Log::TRC) << "parent node: " << parent->nodeId().toFullString().toUtf8();
 
-        const std::list<UaNode::ReferencedTarget>* referenced =  to->referencedTargets();
-        LOG(Log::TRC) << "Referenced nodes: " << referenced->size();
+    const std::list<UaNode::ReferencedTarget>* referenced =  to->referencedTargets();
+    LOG(Log::TRC) << "Referenced nodes: " << referenced->size();
 
-        UA_Argument *inArgs = 0;
-        int inArgsSize = 0;
-        UA_Argument *outArgs = 0;
-        int outArgsSize = 0;
+    UA_Argument *inArgs = 0;
+    int inArgsSize = 0;
+    UA_Argument *outArgs = 0;
+    int outArgsSize = 0;
 
-        for ( std::list<UaNode::ReferencedTarget>::const_iterator it = referenced->cbegin(); it!=referenced->cend(); it++ )
+    for ( std::list<UaNode::ReferencedTarget>::const_iterator it = referenced->cbegin(); it!=referenced->cend(); it++ )
```

Comment:
same sort of thing here re: loop iteration, could also be C++11 range based for?